### PR TITLE
Set minimum pytorch version to 1.8 + cleanup

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes https://github.com/Lightning-AI/metrics/issues/1262
Sets minimum pytorch version to 1.8 and clean-up all skipping/conditioning throughout the code that was necessary to run for older versions.

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by stancld — APPROVED** [review]

**Comment by stancld** [comment]
#1259 has set `v0.10` milestone (maybe for a possible patch release). We should make this in sync.

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 numpy>=1.17.2
-torch>=1.3.1
+torch>=1.8
```
Comment (Borda) [inline requirements.txt:2]:
```suggestion
torch>=1.8.0
```

**Review by Borda — APPROVED** [review]

File: src/torchmetrics/utilities/data.py [hunk 2]
```diff
@@ -14,26 +14,9 @@
 from typing import Any, Callable, Dict, List, Mapping, Optional, Sequence, Union
 
 import torch
-from torch import Tensor, tensor
-
-from torchmetrics.utilities.imports import (
-    _TORCH_GREATER_EQUAL_1_6,
-    _TORCH_GREATER_EQUAL_1_7,
-    _TORCH_GREATER_EQUAL_1_8,
-    _TORCH_GREATER_EQUAL_1_12,
```
[the remark this answers is not in the document]
Comment (SkafteNicki) [reply src/torchmetrics/utilities/data.py:23]: The diff just makes it look weird, the import is still there:
https://github.com/Lightning-AI/metrics/blob/2791e04aacf569463e3c3f0f9de6c0ec82442cc3/src/torchmetrics/utilities/data.py#L19

**Comment by SkafteNicki** [comment]
> #1259 has set `v0.10` milestone (maybe for a possible patch release). We should make this in sync.

You are right. But then we probably need to move #1259 to v0.11 because this kind of breaking change to minimum requirement i think we need to do as an v0.x.0 release and not v0.10.x.

**Comment by SkafteNicki** [comment]
seems like CI pulling the wrong version of torch for `oldest` config. Maybe the cache is needs to be reset?

**Comment by Borda** [comment]
> There's a `.github/assistant.py` file containing the following dict:

seems like a bug in the assistant as it shall take the max

**Comment by Borda** [comment]
@SkafteNicki seems there are a few tests failing for Windows and PT 1.8

**Pull request merged**